From 87addd437c17d76d1b0476881e96769e581b0b75 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 19 Sep 2017 18:47:13 +0800
Subject: [PATCH 2577/5242] MLK-16536-3 video: fbdev: dcss: replace
 'kfifo_esize()' by 'esize'

commit  45e37a681930211578d3c56f2504a76ce609a1eb from
https://source.codeaurora.org/external/imx/linux-imx.git

When doing ctxld config, the 'esize' variable is already
assigned to the value derived from 'kfifo_esize()'. So
using the exsiting value instead of deriving it again.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 37e0e0c..c10c9a6 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2250,7 +2250,7 @@ static void dcss_ctxld_config(struct work_struct *work)
 	/* configure sb buffer */
 	if (cc->sb_data_len) {
 		/* cfifo first store sb and than store db */
-		writel(cfifo->dma_handle + offset * kfifo_esize(&cfifo->fifo),
+		writel(cfifo->dma_handle + offset * esize,
 		       info->base + chans->ctxld_addr + CTXLD_SB_BASE_ADDR);
 		writel(cc->sb_hp_data_len |
 		       ((cc->sb_data_len - cc->sb_hp_data_len) << 16),
@@ -2259,9 +2259,7 @@ static void dcss_ctxld_config(struct work_struct *work)
 
 	/* configure db buffer */
 	if (cc->db_data_len) {
-		writel(cfifo->dma_handle +
-		       (offset + cc->sb_data_len) *
-		       kfifo_esize(&cfifo->fifo),
+		writel(cfifo->dma_handle + (offset + cc->sb_data_len) * esize,
 		       info->base + chans->ctxld_addr + CTXLD_DB_BASE_ADDR);
 		writel(cc->db_data_len,
 		       info->base + chans->ctxld_addr + CTXLD_DB_COUNT);
@@ -2278,7 +2276,7 @@ static void dcss_ctxld_config(struct work_struct *work)
 
 	ctxld_fifo_info_print(cfifo);
 	kfifo_dma_out_finish(&cfifo->fifo,
-		(cc->sb_data_len + cc->db_data_len) * kfifo_esize(&cfifo->fifo));
+		(cc->sb_data_len + cc->db_data_len) * esize);
 	ctxld_fifo_info_print(cfifo);
 
 	kfree(cc);
-- 
1.7.9.5

